cpp/src/scalar/scalar.cpp

```diff
@@ -114,11 +114,10 @@ string_scalar::operator std::string() const { return this->to_string(cudf::get_d
 
 std::string string_scalar::to_string(rmm::cuda_stream_view stream) const
 {
-  std::string result;
-  result.resize(_data.size());
-  CUDF_CUDA_TRY(
-    cudaMemcpyAsync(&result[0], _data.data(), _data.size(), cudaMemcpyDefault, stream.value()));
-  stream.synchronize();
+  std::string result(size(), '\0');
```

Comment:
Is this ctor better than the `.resize()` pattern?

Reply:
Seems ever so slightly better since we don't create an empty string first. The ultimate result is the same. Also, it's one line better :)
I can revert if you prefer the old version.

Reply:
I'm ok with it. Just curious.